### DISCUSSION TITLE
What are the best schema management tools, in your opinion?

### DISCUSSION THREAD

**Discussion by koheiyamayama** [cockroachdb/cockroach#127143] [Q&A]
Hi everyone,

I've been exploring various schema management tools for use with CockroachDB and have tried the following so far:
- Flyway
- psqldef
- Liquibase
- Atlas

All of these tools are very good, but I encountered some limitations:
- Some tools don't support CockroachDB.
- Others don't support undo commands in their community editions.

I would love to hear your recommendations and experiences with schema management tools that work well with CockroachDB. Any suggestions or insights would be greatly appreciated!

**Comment by dikshant** [comment]
Flyway and Liquibase work well with CockroachD and I would encourage you to use those. Can you clarify what you mean by undo commands? Are you referring to undoing the migration? There is also [go-migrate](https://github.com/golang-migrate/migrate) you can try.
